pkg/cli/debug/debug.go

```diff
@@ -170,13 +172,14 @@ func NewDebugOptions(streams genericclioptions.IOStreams) *DebugOptions {
 	attachOpts.TTY = true
 	attachOpts.Stdin = true
 	return &DebugOptions{
-		PrintFlags:         genericclioptions.NewPrintFlags("").WithTypeSetter(scheme.Scheme),
-		IOStreams:          streams,
-		Timeout:            15 * time.Minute,
-		KeepInitContainers: true,
-		AsUser:             -1,
-		Attach:             *attachOpts,
-		LogsForObject:      polymorphichelpers.LogsForObjectFn,
+		PrintFlags:             genericclioptions.NewPrintFlags("").WithTypeSetter(scheme.Scheme),
+		IOStreams:              streams,
+		Timeout:                15 * time.Minute,
+		DebugPodRunningTimeout: 30 * time.Second,
```

Comment:
i'm not sure what time limit is appropriate to wait for a pod status of PodRunning, but in case of this bug, pod immediately fails so cmd immediately exits.